### ISSUE TITLE
How to become a member ? 

### ISSUE THREAD

**Issue by mulle-nat** [mulle-objc/mulle-objc.github.io#1]
There are multiple stages of membership on github, that doesn't really have a non-technical membership per se:

Stage                            | Description
---------------------------|----------------------------------------------
Supporter                    | Star a repository. That's the easiest way to show that you care :)
Community Member   | Make yourself known and give feedback by opening and commenting on issues or asking questions. That makes you a community member. Thank you.
Contributor | You submitted an accepted pull-request. Nice.
Collaborator | Your pull-requests are accepted anyway. It'd be easier for all, if you became a collaborator. Ask for it.
Technical Member | Where you can create repositories and such. You know it yourself, when to ask for it.

**Comment by Maximus-** [comment]
Interested :-)

**Comment by vtsingaras** [comment]
Me too.

**Comment by thecatalinstan** [comment]
I'm  also very interested.

**Comment by jeethu** [comment]
I'm interested!

**Comment by mulle-nat** [comment]
I am using the Heart as a marker to remember sent invitations

**Comment by mvasilkov** [comment]
Please inv, I'm very interested in Obj-C.

**Comment by jrc** [comment]
Let's #MakeObjCGreatAgain!

**Comment by g3ntleman** [comment]
What a great effort. Thought about similar things. Please invite me.

**Comment by apple4ever** [comment]
Super interested!

**Comment by apple4ever** [comment]
Super interested!

**Comment by ardiefox** [comment]
What a fantastic project! I would like to be a part of it!

With the emergence of Swift, people are acting weirdly and started to treat Objective-C as a soon-to-be-dead language or whatsoever, which is a complete nonsense!
But, I cannot deny that since the release of Swift I wanted Apple to release Objective-C to public as well, as a cross-platform, just like Swift. I looked into Cocotron*, GNUSTEP*, but they were somewhat disappointing so I was doing some hobby project called objc2csharp; and now we have this wonderful project.

> I am probably off the Apple platform rather sooner than later, but I don't really like anything else besides C and ObjC [...]

Indeed sir, indeed.

*) I know that they are mere re-implementation of Foundation and Cocoa.What a fantastic project! I would like to be a part of it!

With the emergence of Swift, people are acting weirdly and started to treat Objective-C as a soon-to-be-dead language or whatsoever, which is a complete nonsense!
But, I cannot deny that since the release of Swift I wanted Apple to release Objective-C to public as well, as a cross-platform, just like Swift. I looked into Cocotron*, GNUSTEP*, but they were somewhat disappointing so I was doing some hobby project called objc2csharp; and now we have this wonderful project.

> I am probably off the Apple platform rather sooner than later, but I don't really like anything else besides C and ObjC [...]

Indeed sir, indeed.

*) I know that they are mere re-implementation of Foundation and Cocoa.

**Comment by mulle-nat** [comment]
@ardiefox Lets make the best out of it. My feeling is that Cocotron and GNUStep are somewhat content with catching up with whatever Apple is doing and not doing anything more, but **mulle-objc** is a chance to shape the language ourselves.

**Comment by thecatalinstan** [comment]
@ardiefox, @mulle-nat I've been looking forward to the day when I would be able to do my web work with objective-c and have it deployed on Linux as well as OS X. I took care of the first part (through criollo and its preceding project) and I can't wait to be able to make the first one a reality as well. 

I've also looked into Cocotron and GNUStep (and event went as far as trying Darling). This looks to me like the way forward.

> I don't really like anything else besides C and ObjC

#MakeObjCGreatAgain

**Comment by florianbuerger** [comment]
Oh that is interesting. Would love to be part of it.

**Comment by mulle-nat** [comment]
Public members got an invite to slack, if they have public email addresses. If they don't they should send me an [email](mailto:nat-github-slack@mulle-kybernetik.com).

Those that got an email are markered with a thumbs up. (I wanted to use the hooray, but I made a mistake somewhere :))

**Comment by wprice** [comment]
Interested in joining as a member.

**Comment by psing008** [comment]
Would love to dive into Clang :)

**Comment by vtsingaras** [comment]
I don't see the slack invite. My email is vyronas@vtsingaras.me



[image: --]

Vyronas Tsingaras
[image: http://]vtsingaras.me
<http://vtsingaras.me/?promo=email_sig&utm_source=email_sig&utm_medium=email_sig&utm_campaign=external_links>

**Comment by mulle-nat** [comment]
@vtsingaras 

Checklist :)

Are you a member ? 
Are you a public member ?

Answer to both is NO, so no slack invite yet.

**Comment by vtsingaras** [comment]
vtsingaras commented 6 days ago
Me too. <- I had commented. How do I become a member?

**Comment by mulle-nat** [comment]
I sent you an invite via GitHub, you should have gotten a  mail or a notification. On the top right of your GH screen there is a bell. When you click on it, it should be there amongst the messages, I would assume. I can't make you a member myself. You have to accept the invite.

**Comment by vtsingaras** [comment]
Strange, I searched my emails for emails originated from GitHub and didn't have any fo mulle-objc. Anyway, now I saw the invite and have accepted it, thanks!

**Comment by mulle-nat** [comment]
Now you just have to become public :) [HOWTO: become a public member](https://help.github.com/articles/publicizing-or-hiding-organization-membership/)

**Comment by vtsingaras** [comment]
Done :)

**Comment by daniel-beard** [comment]
Please add me as well.

**Comment by 3a4oT** [comment]
ready to explore

**Comment by josh64x2** [comment]
I'd love to join in the fun :)

**Comment by JX7P** [comment]
I'd be very interested.

**Comment by serjepatoff** [comment]
Please add me too!

**Comment by clozinski** [comment]
I am interested in becoming a member, and in writing a Python interpreter in Mulle-Objc.
https://blogory.org/mulle-objc

**Comment by mulle-nat** [comment]
I have the feeling, that the invites are sometimes not properly sent by github. So in addition to the heart, i will reply here and CC the outstanding requests in this thread like so:

Pending Invitations |  
--------------------|
@clozinski               |
@serjepatoff           |

**Comment by clozinski** [comment]
It was quite difficult to join the organizatoin. 
You want to add instructions at the top of this page. 
First post here, you got that part, then when invited go to 
https://github.com/mulle-objc
click on the button on the right to join. 

The Github instructions were terrile.  I thought I had to go to my home page to join this oranization.

**Comment by mulle-nat** [comment]
I would like to put something directly on https://github.com/mulle-objc, but I think I can only list repositories there. I reworded the top blurb in this issue to make the process clearer hopefully. Is it better now ?

**Comment by clozinski** [comment]
I am still quite confused about how the process works. 

I would say something like the following. 
To become a member, just post on this issue.  I will send you an email invitation to both this organization, and to the slack channel.  Just click on the buttons and you are done. 

(Editor's note: Otherwise new people spend way way too much time clickign around trying to figure out what to do. If I were not motivaed, I would have given up earlier. )

**Comment by mulle-nat** [comment]
Yeah but that changes the semantics of the game. Because people are then just private, There is no real tangible benefit for being a private member of mulle-objc for both sides at the moment.

People, who want to ask a question or discuss something quickly, can just put up a new issue here.

**Comment by clozinski** [comment]
Well that looks better.  Here is what you want to add.
 
So becoming a member is a three stage process.  First add a comment on this issue tracker.  
If your email is visible on github, I will send you an email.  Click on the button, and you become a private member.   Then go to this web page
https://github.com/orgs/mulle-objc/people
Find your name in the first column, and in the second column change your security setting to public. 
I will then invite you to the slack user group.

For those who have done this on github before, not a problem, for the new user, what a pain.  But I do want to make it easier for those who follow in my footsteps.

And the github docs say my photo is on the right.  In fact it is on the left.  Just makes me not trust the docs.  
So do not mention them, just give the person the URL to the list of people.  Saves much effort.  The more a person has to click , the bigger chance of loosing them. 

Hope that helps.

**Comment by mulle-nat** [comment]
I updated the blurb with your suggestions. I think it looks good now.

**Comment by mtecza** [comment]
I followed your Optimising Obj-C series many years ago, and too like Obj-C to live.

**Comment by mulle-nat** [comment]
Thx for joining. @mtecza If you want slack access you need to become a public member (see first post).

**Comment by codeon-nat** [comment]
Thanks. Appreciate the sentiment!

**Comment by markdhughes** [comment]
Hook me up, I need to get back to coding in a sane language!

**Comment by mulle-nat** [comment]
You've come to the right place, probably 😃 ...

**Comment by hsnamr** [comment]
I never asked for Swift. All I ever wanted was Objective-C 3.0

#MakeObjCGreatAgain

**Comment by jkoutavas** [comment]
Bring me to the next level, @mulle-nat :-)

**Comment by mulle-kybernetik-tv** [comment]
@jkoutavas thanks for becoming a public member, if you want a slack invite I need an email-address somehow.

**Comment by jkoutavas** [comment]
I sent you an email at the time I signed up. I’ll resend when I get back at my computer.

**Comment by Belomi** [comment]
I quite like Objective-C and am interested in seeing where the language can go.

**Comment by bugravirgomercury** [comment]
I too want to be a part of it. I'll write a game with it even.

**Comment by mulle-nat** [comment]
A game written in mulle-objc would be very cool indeed.

**Comment by mulle-nat** [comment]
@bugravirgomercury if you want an invite for Slack send me an email to [nat-github-slack@mulle-kybernetik.com](mailto:nat-github-slack@mulle-kybernetik.com)

**Comment by alepi** [comment]
Hi, I am interested in porting mulle objc ecosystem to openbsd and I need some guidance for that.
For example, openbsd makes its own patches to llvm and clang.
Is the community the right place to ask this kind of questions? :-)

**Comment by mulle-nat** [comment]
@alepi if you want an invite for Slack send me an email to nat-github-slack@mulle-kybernetik.com

**Comment by 8bitcynic** [comment]
Hello, could I join your project? I am currently working on a SAS compiler and am looking at Objective-C as an upgrade to the C code I am working with, and am excited to see where mulle-objc will go.

**Comment by mulle-nat** [comment]
Sure ;) What's SAS ?

**Comment by 8bitcynic** [comment]
> Sure ;) What's SAS ?

Thank you for the invitation! SAS is a programming language used for statistics, in a similar vein as the R language.

**Comment by mulle-nat** [comment]
Cool! I guess yours it not open source ? I didn't see anything on your account. Other Objective-C libraries are always interesting to look at.

The community is here basically. There is no mailing list, but there is a  "Slack", which is very quiet. In terms of getting help or feedback, you can try various channels, like github issues on any related project, Slack posts, emails to me. 
*To get access to Slack you got to run through the low-obstacle course pointed out in the first post.*

Just in case things don't work out, negative feedback would also be appreciated. Maybe even more appreciated, than positive feedback, because I already know that it's good :laughing:.

**Comment by minusbat** [comment]
Oh, it's open source, and you can get it here if you want to take a look...

git://toybox.twisted.org.uk/pfkit

Am slightly cringing at the idea of someone seeing that code however :-)

[ I originally wrote it back in 1999, so it was under RCS, and later SVN. I did give it out on mailing lists a couple of times. But as its pretty specialised to me (and even more so 23 years later) then I never really made a thing of it. I did make a git repository though, as it makes it easier for work, and that's what the link is above. ]

So, yes, I shall have a go with your runtime as soon as I get some spare time, and shall give you feedback! Is it a drop-in replacement for libobjc2, or will I need to tweak some calls ? Am assuming the latter, plus I also note you seem to have your own fork of clang ? Do I need to use that ?

Anyway, thanks for the fast reply!

**Comment by minusbat** [comment]
Am glad it didn't horrify you too much ;) Ok, I will start by downloading and building your clang fork then. I found your compatibility layer (option 2) and thought that would be a good start too. Ideally I would ut n the effort to go fully option 1 though.

Option 3 is interesting. The problem might be the 300,000+ lines of commnercial code which sits on top of all of this, and which might have dependencies on odd behaviours of my PFObject that I am unaware of (I wrote all of that too, its been a long 23 years).

But I shall start with the compiler and see what I can get running... thanks for the tips!

**Comment by mulle-nat** [comment]
Yeah, good luck! I think in terms of highlevel runtime differences, there are two things to keep in mind. 

* the mulle-objc-runtime was written to be lock free and wait free, this isn't super-true anymore due to the way `+initialize` works. Yet: one of the consequences of the design are, you can only add methods and classes, you can not remove anything.
*  the mulle-objc-runtime uses the [metaABI](https://www.mulle-kybernetik.com/weblog/2015/mulle_objc_meta_call_convention.html) for method calls. So doing method calls "outside" of the mulle-clang compiler   for example using `IMP` is very different.

Hit me up anytime something comes up.

**Comment by mulle-nat** [comment]
@minusbat So how did it go ? I assume there were no problems, since I didn't hear anything :smile:

**Comment by strojacek** [comment]
Hello, if you are still accepting members, could I join the group?

**Comment by mulle-nat** [comment]
Sure thing. An invitation should be at your doorstep.

**Comment by stianhoiland** [comment]
This project is amazing, wow! You're a wizard, mulle. Please send me an invite to join :) As an aside: Would you consider bridging the chat to Discord?

**Comment by mulle-nat** [comment]
Thanks for the kind words. An invite has been sent. I would consider bridging the Slack to anything, but there is just not enough traffic there to make it worthwhile the effort.

**Comment by stianhoiland** [comment]
Getting a Discord server up would lower the bar for initiating contact, potentially on-boarding fence-sitters. And can I just say, this project needs more attention. 

Also, I've done the public member dance thingy. *waits impatiently*

**Comment by mulle-nat** [comment]
The Slack invite has been sent a few hours ago, since you are waiting impatiently, you may want to check your SPAM folder :smile:

**Comment by cleanbaja** [comment]
@mulle-nat This project is very interesting, and I would love to join (if this project hasn't yet died)

**Comment by mulle-nat** [comment]
The project is alive and well and has no signs of slowing down :smile:

**Comment by mulle-nat** [comment]
@cleanbaja If you email [me](mailto:nat-invite@mulle-kybernetik.com) your email address. I can send you a slack invite.

**Comment by wenq1** [comment]
Ummmm

**Comment by johnsonjh** [comment]
Heyooo o/

**Comment by elisauhura** [comment]
I would like to join the project too :)

**Comment by KCommit** [comment]
I would like to join the project too :)

**Comment by chazu** [comment]
Hey would love to join! Stoked to learn about this project :D

**Comment by aforgiel** [comment]
Allo,
I'm interested as well !

**Comment by mejbass** [comment]
hello, 
I'm Interested too.

**Comment by mulle-nat** [comment]
The slack has migrated to discord. The invite is in the top post.
